### DISCUSSION TITLE
Shared cache for libraries used by multiple projects

### DISCUSSION THREAD

**Discussion by ronw98** [ccache/ccache#1530] [Questions & answers]
I was wondering if the same ccache could be shared across different projects. In my specific case I have a white label mobile app with many different ios schemes and they seem to be using different caches (building a new scheme nevers hits anything in cache but building a previously cached scheme hits).
The code is the exact same across all the schemes the only difference are the assets such as app icon and launch screen.

Is there a way to configure ccache so that it uses the same cache for all the different schemes ?

In case it is useful:
```text
ccache -p

(default) absolute_paths_in_stderr = false
(default) base_dir = 
(default) compiler = 
(default) compiler_check = mtime
(default) compiler_type = auto
(default) compression = true
(default) compression_level = 0
(default) cpp_extension = 
(default) debug = false
(default) debug_dir = 
(default) debug_level = 2
depend_mode = true
(default) direct_mode = true
(default) disable = false
(default) extra_files_to_hash = 
file_clone = true
(default) hard_link = false
(default) hash_dir = true
(default) ignore_headers_in_manifest = 
(default) ignore_options = 
inode_cache = true
(default) keep_comments_cpp = false
(default) log_file = 
(default) max_files = 0
max_size = 40.0 GB
(default) msvc_dep_prefix = Note: including file:
(default) namespace = 
(default) path = 
(default) pch_external_checksum = false
(default) prefix_command = 
(default) prefix_command_cpp = 
(default) read_only = false
(default) read_only_direct = false
(default) recache = false
(default) remote_only = false
(default) remote_storage = 
(default) reshare = false
(default) run_second_cpp = true
sloppiness = clang_index_store, file_stat_matches, include_file_ctime, include_file_mtime, ivfsoverlay, modules, pch_defines, system_headers, time_macros
(default) stats = true
(default) stats_log = 
(default) umask = 
```

I have seen [this discussion](https://github.com/ccache/ccache/discussions/786) but don't know if it is the exact same topic or not.
Any help would be greatly appreciated!

**Comment by jrosdahl** [comment]
It's not that different caches are used for different schemes, it's that the so-called "input hash" (essentially "what does the source code and its environment look like?") is different which leads to different cache keys (i.e., cache misses). To find out what the difference is you'll have to enable [debug logging](https://ccache.dev/manual/4.10.2.html#_cache_debugging) and analyze the result.

I don't know how iOS builds work, but if different schemes build into different directories then the discussion you mention is likely relevant. It could also be the case that the build system sets up scheme-specific arguments, in which case it might or might not be possible to share cache entries.
